runtime/lua/vim/ui.lua

```diff
@@ -103,4 +103,46 @@ function M.input(opts, on_confirm)
   end
 end
 
+--- Prompts the user to pick an option from a list of choices
+---
+---@param msg string Prompt displayed to the user
+---@param opts table
+---     - choices (string[]|nil)
+---               List of choices. The shortcut key for each choice is designated with a "&" character. If "&" is not present in the choice, the first letter of the choice is used.
+---		  See |confirm()| for additional informations.
+---     - default (number|nil)
+---               Default choice selected when user presses <Enter>
+---@param on_choice fun(idx:number|nil,choice:string|nil)
+---               Called once the user chooses or cancel the input.
+---               `idx` is the 1-based index of the 'choice' in `opts.choices` table,
+---               (it might be `0` if none of the choices is picked, or
+---               `nil` if the user canceled the dialog.)
+---		  `choice` is what the user chose (`nil` on cancelation or
+---		  if user did not choose any option)
+---
+--- Example:
+--- <pre>
+--- vim.ui.confirm("Favorite fruit?",{
+---     choices = {'Apple', 'Ba&nana', 'Cramberry' },
+---     default = 2
+--- }, function(idx, choice)
+---     print(idx, choice)
+--- end)
+--- </pre>
+function M.confirm(msg, opts, on_choice)
+  vim.validate({
+    msg = { msg, 'string' },
+    opts = { opts, 'table', true },
+    on_choice = { on_choice, 'function' },
+  })
+  opts = opts or {}
+  opts.choices = opts.choices or {}
+  local ok, choice_idx = pcall(vim.fn.confirm, msg, table.concat(opts.choices, '\n'), opts.default)
+  if not ok then
+    on_choice(nil, nil)
+  else
+    on_choice(choice_idx, (opts.choices[choice_idx]:gsub('&', '')))
+  end
```

Comment:
Proper error handling. `confirm` returns `0` on Esc and `Keyboard interrupt` error on Ctrl-C
```suggestion
  local ok, res = pcall(vim.fn.confirm, msg, table.concat(opts.choices, '\n'), opts.default)
  if (ok and res == 0) or (not ok and res == 'Keyboard interrupt') then
    on_choice(nil, nil)
  elseif ok then
    on_choice(res, (opts.choices[res]:gsub('&', '')))
  else
    error(res)
  end
```